Helper/Api/Attributes.php

```
@@ -480,4 +496,33 @@ public function getAttributeSetIdByName(string $catalogEntityId, string $attribu
 
         return $attributeSetId;
     }
+
+    /**
+     * Check if attribute is attached to a certain attribute set
+     *
+     * @param string $attributeSetId
+     * @param string $attributeCode
+     * @param string $entityTypeCode
+     * @return bool
+     */
+    public function isAttributeInAttributeSet(
+        string $attributeSetId, string $attributeCode, string $entityTypeCode = Product::ENTITY
+    ) {
+        try {
+            $attributes = $this->attributeManagement->getAttributes(
+                $entityTypeCode,
+                $attributeSetId
+            );
+
+            foreach ($attributes as $attribute) {
+                if ($attribute->getAttributeCode() == $attributeCode) {
+                    return true;
+                }
+            }
+        } catch (\Exception $e) {
+            $this->_logger->error($e->getMessage());
```

what can go wrong here? Shouldn't let it fail if it does?